### PR TITLE
Add ball throwing

### PR DESCRIPTION
**Pull request by lcaonmst**
Heuristic approach to throw ball into the paddle from nearby locations. Transform the distance vector into force vector by some simple transposition.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -38,6 +42,13 @@ def get_mode():
 
     update_wind_controllers(p, *wind_controllers)
 
+    # Check if the force button was clicked, and throw the ball eventually.
```
Comment (matiutd888) [inline main.py:45]: "Eventually" doesn't mean "ewentualnie". I don't understand why this word is used in the comment.

File: main.py [hunk 2]
```diff
@@ -27,6 +29,8 @@ def get_mode():
 else:
     paddle.create_joint_controllers()
 
+throwBallButton = Button(p.addUserDebugParameter("Throw ball towards paddle", 1, 0, 0))
```
Comment (matiutd888) [inline main.py:32]: Please use snake_case in the names of variables (throw_ball_button).

**Review by matiutd888 — COMMENTED** [review]
Still will need to come up with a non-heuristic approach to the throwing problem. Nevertheless, a heuristic one suffices for now.

**Review by matiutd888 — APPROVED** [review]

File: paddle/paddle.py [hunk 3]
```diff
@@ -71,6 +72,11 @@ def move_to_position(self, p, vel=1):
                                                        targetPosition=p[i],
                                                        maxVelocity=vel)
 
+
+    def get_center_position(self):
```
Comment (Barrus00) [inline paddle/paddle.py:76]: Między metodami w klasie powinna być jedna linia wcięcia.

File: utils/physics.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    return tuple(map(operator.add, v1, v2))
+
+def sub_vertices(v1, v2):
+    assert len(v1) == len(v2)
+    return tuple(map(operator.sub, v1, v2))
+
+# Finds recommended force needed to let ball travel some distant.
+def scale_vector(v):
+    assert len(v) == 3
+    return (1.3 * v[0], 1.3 * v[1], v[2] + 2)
```
Comment (Barrus00) [inline utils/physics.py:15]: Magiczne stałe... kiedyś nas to zgubi.

**Review by Barrus00 — CHANGES_REQUESTED** [review]

**Review by Barrus00 — APPROVED** [review]

**Pull request merged**